MatrixSDK/MXSession.m

```diff
@@ -1974,8 +1982,269 @@ - (MXHTTPOperation*)leaveGroup:(NSString*)groupId
     } failure:failure];
 }
 
+- (MXHTTPOperation*)updateGroupProfile:(MXGroup*)group
+                            success:(void (^)(void))success
+                            failure:(void (^)(NSError *error))failure
+{
+    if (!group.groupId.length)
+    {
+        return nil;
+    }
+    
+    NSLog(@"[MXSession] updateGroupProfile %@", group.groupId);
+    
+    __weak __typeof(self)weakSelf = self;
+    
+    return [matrixRestClient getGroupProfile:group.groupId success:^(MXGroupProfile *groupProfile) {
+        
+        if (weakSelf)
+        {
+            typeof(self) self = weakSelf;
+            MXGroup *storedGroup = [self groupWithGroupId:group.groupId];
+            
+            if (storedGroup != group)
+            {
+                // Update the provided group instance
+                group.profile = groupProfile;
+            }
+            
+            if (storedGroup && [storedGroup updateProfile:groupProfile])
+            {
+                [_store storeGroup:storedGroup];
+                // Commit store changes done in [room handleMessages]
```

Comment:
The comment does not correspond to the current case.
Note: it badly appears several times in this PR.